### ISSUE TITLE
SessionID (2nd module parameter) is a list in "websocket" controllers, and a binarry in "controller" controllers

### ISSUE THREAD

**Issue by chatterbeak** [ChicagoBoss/ChicagoBoss#632]
The module parameter "SessionID" comes into websocket controllers as a binary, for example

<<"1289ba634f9d1736a9f93dba3305c82917f8d037">>

but it's passed into HTTP/rest controllers as a list, for example:

"1289ba634f9d1736a9f93dba3305c82917f8d037"

They should be consistent, I presume as a list.

**Comment by choptastic** [comment]
So, it appears that the websocket handler [converts the session_key to a binary](https://github.com/ChicagoBoss/ChicagoBoss/blob/master/src/boss/boss_simple_bridge_handler.erl#L34-L36) (which ensures that the value of the cookie will be returned as a binary).

But the boss_web_controller does [not do that](https://github.com/ChicagoBoss/ChicagoBoss/blob/master/src/boss/boss_web_controller.erl#L219-L220).

To make it consistent (assuming you want it to be a binary always), I'd have `boss_session:get_session_key()` either always return a binary, that way the return for both the above calls to `Bridge:cookie/1` will be consistent, and you won't have to do any explicit casting.

But it all depends on how dependent folks are on Sessionid explicitly being a list in their controllers.  Making the WS handler use lists will likely break fewer things.

**Comment by chatterbeak** [comment]
The reason I suggested "list" was I figured there are more people using the web controllers than the websocket controller (especially since the websocket only works if you're using Cowboy as the web server). If it were up to me, I'd prefer binaries.

**Comment by choptastic** [comment]
Indeed.  It *could* be user-configurable by having it completely depend on the type set in boss.config.  That way, if the session_key in boss.config is a list, then in your controllers, it's a list, or if it's a binary in boss.config, then it's a binary in the controllers.  Then the list_to_binary conversion would be removed from ws_init in the simple_bridge handler.
